node/src/request_coordinator.ts

```diff
@@ -0,0 +1,65 @@
+import {promises as DNS, SrvRecord} from 'dns';
+import {AdminClient} from './admin_client';
+
+/**
+ * Supports to send pause/unpause requests to all hosts of a service records
+ */
+export class RequestCoordinator {
+  srvServiceUrl: string;
+  resolver: DNS.Resolver;
+
+  constructor(srvServiceUrl: string) {
```

Comment:
Not a problem at all but just sharing info. In Typescript, you can do the following and then you don't have to declare your class instances explicitly.  Depends on what style you like I guess :)

https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/